unity/Assets/Scripts/ImageSynthesis/Shaders/Depth.shader

```diff
@@ -51,24 +56,21 @@ Shader "Hidden/Depth" {
              fixed4 frag(output o) : COLOR
              {
                  // depth01 = pow(LinearEyeDepth(depth01), _DepthLevel);
-                 float depth01 = Linear01Depth(UNITY_SAMPLE_DEPTH(tex2D(_CameraDepthTexture, o.uv)));
-
-                 // float lowBits = depth01 / 2;
-                 // float medBits = depth01 / 4;
-                 // float highBits = depth01 / 8;
+                 float multiplier = _FarClipPlane - _NearClipPlane;
+                 float depth01 = Linear01Depth(UNITY_SAMPLE_DEPTH(tex2D(_CameraDepthTexture, o.uv))) * multiplier;
+                 // This converts the float to an int bitwise (not a cast)
+                 // https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/asint
+                 uint depthUint = asint(depth01);
 
-                 // float depth01 = 1 - UNITY_SAMPLE_DEPTH(tex2D(_CameraDepthTexture, o.uv));
-     			 float lowBits = floor(depth01 * 256) / 256;
-				 float medBits = 256 * (depth01 - lowBits);
-				 medBits = floor(256 * medBits) / 256;
-				 float highBits = 256 * 256 * (depth01 - lowBits - medBits / 256);
-			  	 highBits = floor(256 * highBits) / 256;
+                 float f1 = float(depthUint & 255) / 255.0;
+                 float f2 = float((depthUint >> 8) & 255) / 255.0;
+                 float f3 = float((depthUint >> 16) & 255)  / 255.0;
+                 float f4 = float((depthUint >> 24) & 255) / 255.0;
 
-				 return fixed4(lowBits, medBits, highBits, 0.0);
-                 // return depth01;
+                 return fixed4(f1, f2, f3, f4);
```

Comment:
We may be able to change the shader signature and shader's operating texture format to be a 32bit float to not have to transform to ints. I remember having problems in Metal with something similar so just a note